src/combinatorics/simple_graph/basic.lean

```diff
@@ -94,39 +97,119 @@ by { classical, exact fintype.of_injective simple_graph.adj simple_graph.ext }
-def empty_graph (V : Type u) : simple_graph V :=
-{ adj := λ i j, false }
+/-- The graph with no edges on a given vertex type `V`. `mathlib` prefers the notation `⊥`. -/
+def empty_graph (V : Type u) : simple_graph V := { adj := λ i j, false }
 
 namespace simple_graph
 
 variables {V : Type u} {W : Type v} {X : Type w} (G : simple_graph V) (G' : simple_graph W)
 
+@[simp] lemma irrefl {v : V} : ¬G.adj v v := G.loopless v
+
+lemma adj_comm (u v : V) : G.adj u v ↔ G.adj v u := ⟨λ x, G.sym x, λ x, G.sym x⟩
+
+@[symm] lemma adj_symm {u v : V} (h : G.adj u v) : G.adj v u := G.sym h
+
+lemma ne_of_adj {a b : V} (hab : G.adj a b) : a ≠ b :=
+by { rintro rfl, exact G.irrefl hab }
+
+section order
+
+/-- The relation that one `simple_graph` is a subgraph of another. -/
+def is_subgraph (x y : simple_graph V) : Prop := ∀ ⦃v w : V⦄, x.adj v w → y.adj v w
+
+instance : boolean_algebra (simple_graph V) :=
+{ le := is_subgraph,
+  sup := λ x y,
+        { adj := x.adj ⊔ y.adj,
+          sym := λ v w h, by rwa [sup_apply, sup_apply, x.adj_comm, y.adj_comm] },
+  inf := λ x y,
+        { adj := x.adj ⊓ y.adj,
+          sym := λ v w h, by rwa [inf_apply, inf_apply, x.adj_comm, y.adj_comm] },
+  compl := λ G,
+        { adj := λ v w, v ≠ w ∧ ¬G.adj v w,
+          sym := λ v w ⟨hne, _⟩, ⟨hne.symm, by rwa adj_comm⟩,
+          loopless := λ v ⟨hne, _⟩, (hne rfl).elim },
+  sdiff := λ x y,
+        { adj := x.adj \ y.adj,
+        sym := λ v w h, by change x.adj w v ∧ ¬ y.adj w v; rwa [x.adj_comm, y.adj_comm] },
+  top := complete_graph V,
+  bot := empty_graph V,
+  le_top := λ x v w h, x.ne_of_adj h,
+  bot_le := λ x v w h, h.elim,
+  sup_le := λ x y z hxy hyz v w h, h.cases_on (λ h, hxy h) (λ h, hyz h),
+  sdiff_eq := λ x y, by { ext v w, refine ⟨λ h, ⟨h.1, ⟨_, h.2⟩⟩, λ h, ⟨h.1, h.2.2⟩⟩,
+                          rintro rfl, exact x.irrefl h.1 },
+  sup_inf_sdiff := λ a b, by { ext v w, refine ⟨λ h, _, λ h', _⟩,
+                               obtain ⟨ha, _⟩|⟨ha, _⟩ := h; exact ha,
+                               by_cases b.adj v w; exact or.inl ⟨h', h⟩ <|> exact or.inr ⟨h', h⟩ },
+  inf_inf_sdiff := λ a b, by { ext v w, exact ⟨λ ⟨⟨_, hb⟩,⟨_, hb'⟩⟩, hb' hb, λ h, h.elim⟩ },
+  le_sup_left := λ x y v w h, or.inl h,
+  le_sup_right := λ x y v w h, or.inr h,
+  le_inf := λ x y z hxy hyz v w h, ⟨hxy h, hyz h⟩,
+  le_sup_inf := λ a b c v w h, or.dcases_on h.2 or.inl $
+    or.dcases_on h.1 (λ h _, or.inl h) $ λ hb hc, or.inr ⟨hb, hc⟩,
+  inf_compl_le_bot := λ a v w h, false.elim $ h.2.2 h.1,
+  top_le_sup_compl := λ a v w ne, by { by_cases a.adj v w, exact or.inl h, exact or.inr ⟨ne, h⟩ },
+  inf_le_left := λ x y v w h, h.1,
+  inf_le_right := λ x y v w h, h.2,
+  .. partial_order.lift adj ext }
+
+@[simp] lemma sup_adj (x y : simple_graph V) (v w : V) :
+(x ⊔ y).adj v w ↔ (x.adj v w ∨ y.adj v w) := iff.rfl
+
+@[simp] lemma inf_adj (x y : simple_graph V) (v w : V) :
+(x ⊓ y).adj v w ↔ (x.adj v w ∧ y.adj v w) := iff.rfl
```

Comment:
```suggestion
@[simp] lemma sup_adj (x y : simple_graph V) (v w : V) : (x ⊔ y).adj v w ↔ x.adj v w ∨ y.adj v w :=
iff.rfl

@[simp] lemma inf_adj (x y : simple_graph V) (v w : V) : (x ⊓ y).adj v w ↔ x.adj v w ∧ y.adj v w :=
iff.rfl
```
resolves the line wrapping here. Alternatively:

```suggestion
@[simp] lemma sup_adj (x y : simple_graph V) (v w : V) :
  (x ⊔ y).adj v w ↔ (x.adj v w ∨ y.adj v w) := iff.rfl

@[simp] lemma inf_adj (x y : simple_graph V) (v w : V) :
  (x ⊓ y).adj v w ↔ (x.adj v w ∧ y.adj v w) := iff.rfl
```

The lemma statement should always be indented if it is wrapped.